### ISSUE TITLE
Reproducing Results

### ISSUE THREAD

**Issue by nilsleh** [bjbbbb/Conditional-Optimization-for-Adaptive-Thresholding#1]
Hi @bjbbbb ,

Thank you for the interesting work. I am trying to adapt the COAT method to include it in the [lightning-uq-box](https://github.com/lightning-uq-box/lightning-uq-box) and am in the process of trying to reproduce the results you reported in the paper. Do you have the model checkpoints available somewhere for me to test the COAT method?

**Comment by bjbbbb** [comment]
Dear nilsleh,

Thank you for your interest in COAT and for working on its integration into lightning-uq-box.
 
Unfortunately, I no longer have the pretrained model checkpoints available. The experiments were conducted on a shared server with limited GPU memory and storage resources. During a cleanup in May 2026, the previous experimental files and model checkpoints were removed to make room for other experiments.
 
Therefore, I am currently unable to provide the original checkpoints or the corresponding calibration values. I sincerely apologize for the inconvenience. The code and the experimental details in the paper should still provide a basis for reproducing the results, although the exact numerical results may vary slightly depending on the data split, training environment, and random seed.

Best regards,


Jie


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "bjbbbb/Conditional-Optimization-for-Adaptive-Thresholding"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2026年9月19日(星期六) 凌晨1:17
***@***.***&gt;;
抄送:&nbsp;"Jie ***@***.******@***.***&gt;;
主题:&nbsp;[bjbbbb/Conditional-Optimization-for-Adaptive-Thresholding] Reproducing Results (Issue #1)



nilsleh created an issue (bjbbbb/Conditional-Optimization-for-Adaptive-Thresholding#1)
 
Hi @bjbbbb ,
 
Thank you for the interesting work. I am trying to adapt the COAT method to include it in the lightning-uq-box and am in the process of trying to reproduce the results you reported in the paper. Do you have the model checkpoints available somewhere for me to test the COAT method?
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications, keep track of coding agent tasks and review pull requests on the go with GitHub Mobile for iOS and Android. Download it today!

**Comment by nilsleh** [comment]
Thank you for your reply. If the checkpoints are not available, can you point me to the code implementation of models you were using? For example what kind of UNet etc? I can run the experiments myself but there are a lot of UNet variants out there for example and I would want to try and reproduce your work correctly.

**Comment by bjbbbb** [comment]
I think it does not matter if you ask any AI to implement UNet, since the method in this paper is compatible with most segmentation models. It is worth noting that, similar to the AA-CRC approach, when training COAT, performing calibration, and evaluating test results, you need to divide the temperature parameter for the sigmoid output of the model by 10 for smoothing.


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "bjbbbb/Conditional-Optimization-for-Adaptive-Thresholding"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2026年9月20日(星期天) 下午2:20
***@***.***&gt;;
抄送:&nbsp;"Jie ***@***.******@***.***&gt;;
主题:&nbsp;Re: [bjbbbb/Conditional-Optimization-for-Adaptive-Thresholding] Reproducing Results (Issue #1)



nilsleh left a comment (bjbbbb/Conditional-Optimization-for-Adaptive-Thresholding#1)
 
Thank you for your reply. If the checkpoints are not available, can you point me to the code implementation of models you were using? For example what kind of UNet etc? I can run the experiments myself but there are a lot of UNet variants out there for example and I would want to try and reproduce your work correctly.
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications, keep track of coding agent tasks and review pull requests on the go with GitHub Mobile for iOS and Android. Download it today!

**Comment by nilsleh** [comment]
Sure but there are different ways of implementing a UNeT for example in the segmentations pytorch library you can pair it with any encoder from timm, so for the purpose of reproducing the results of the paper it would be tremendously helpful to know what UNeT implementation was used.

**Comment by bjbbbb** [comment]
Yes, but I believe the scenario I described does not affect the method in this paper. Even a vanilla UNet can see performance improvements when integrated with my model, and you can test this easily without spending much time. The main issue is that I am not in Hong Kong right now and cannot connect to that server.



---Original---
From: "Nils ***@***.***&gt;
Date: Sun, Sep 20, 2026 22:46 PM
To: ***@***.***&gt;;
Cc: "Jie ***@***.******@***.***&gt;;
Subject: Re: [bjbbbb/Conditional-Optimization-for-Adaptive-Thresholding] Reproducing Results (Issue #1)


nilsleh left a comment (bjbbbb/Conditional-Optimization-for-Adaptive-Thresholding#1)
 
Sure but there are different ways of implementing a UNeT for example in the segmentations pytorch library you can pair it with any encoder from timm, so for the purpose of reproducing the results of the paper it would be tremendously helpful to know what UNeT implementation was used.
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications, keep track of coding agent tasks and review pull requests on the go with GitHub Mobile for iOS and Android. Download it today!

**Comment by nilsleh** [comment]
Yes, I am aware that COAT is model agnostic and can be applied to any segmentation model architecture in this case to get improvements! 

I am only saying that the overall performance of metrics you get (for the purpose of reproducing results in Table 1 for example) depend on the details of the base architecture. 

When I train different UNeT architectures with different model capacity (so one has double the parameters of the other for example) as base models and subsequently apply COAT to them, they might still both benefit, but without knowing what sort of UNeT was used for the experiments in Table 1 for example, it is not possible to truthfully reproduce the reported values. 

I have run a dozen experiments now with different UNeT variants on the POLYP dataset and COAT improves the results for each of them, but none of them exactly (or very closely) reproduce the reported values in the paper. That is why I am asking for more details about the model architecture.

**Comment by bjbbbb** [comment]
I found a PSPNet code on my local computer and hope it will be helpful to you.


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "bjbbbb/Conditional-Optimization-for-Adaptive-Thresholding"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2026年9月21日(星期一) 下午2:21
***@***.***&gt;;
抄送:&nbsp;"Jie ***@***.******@***.***&gt;;
主题:&nbsp;Re: [bjbbbb/Conditional-Optimization-for-Adaptive-Thresholding] Reproducing Results (Issue #1)



nilsleh left a comment (bjbbbb/Conditional-Optimization-for-Adaptive-Thresholding#1)
 
Yes, I am aware that COAT is model agnostic and can be applied to any segmentation model architecture in this case to get improvements!
 
I am only saying that the overall performance of metrics you get (for the purpose of reproducing results in Table 1 for example) depend on the details of the base architecture.
 
When I train different UNeT architectures with different model capacity (so one has double the parameters of the other for example) as base models and subsequently apply COAT to them, they might still both benefit, but without knowing what sort of UNeT was used for the experiments in Table 1 for example, it is not possible to truthfully reproduce the reported values.
 
I have run a dozen experiments now with different UNeT variants on the POLYP dataset and COAT improves the results for each of them, but none of them exactly (or very closely) reproduce the reported values in the paper. That is why I am asking for more details about the model architecture.
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications, keep track of coding agent tasks and review pull requests on the go with GitHub Mobile for iOS and Android. Download it today!

**Comment by bjbbbb** [comment]
This is the model code I kept for other methods. Sorry, the UNet code is on the server and I cannot access it right now.
------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "bjbbbb/Conditional-Optimization-for-Adaptive-Thresholding"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2026年9月21日(星期一) 下午2:21
***@***.***&gt;;
抄送:&nbsp;"Jie ***@***.******@***.***&gt;;
主题:&nbsp;Re: [bjbbbb/Conditional-Optimization-for-Adaptive-Thresholding] Reproducing Results (Issue #1)



nilsleh left a comment (bjbbbb/Conditional-Optimization-for-Adaptive-Thresholding#1)
 
Yes, I am aware that COAT is model agnostic and can be applied to any segmentation model architecture in this case to get improvements!
 
I am only saying that the overall performance of metrics you get (for the purpose of reproducing results in Table 1 for example) depend on the details of the base architecture.
 
When I train different UNeT architectures with different model capacity (so one has double the parameters of the other for example) as base models and subsequently apply COAT to them, they might still both benefit, but without knowing what sort of UNeT was used for the experiments in Table 1 for example, it is not possible to truthfully reproduce the reported values.
 
I have run a dozen experiments now with different UNeT variants on the POLYP dataset and COAT improves the results for each of them, but none of them exactly (or very closely) reproduce the reported values in the paper. That is why I am asking for more details about the model architecture.
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications, keep track of coding agent tasks and review pull requests on the go with GitHub Mobile for iOS and Android. Download it today!

**Comment by nilsleh** [comment]
Thank you, but unfortunately I cannot see the attachement here. Could you kindly send it to my email at n.lehmann@tum.de? And whenever you get access to the server for the UNeT implementation that would be great as well, but no time pressure. Thanks and much appreciated!
